File: frontend/projects/api/src/lib/import-status.ts
import type { CurriculumStatus } from './curriculum-status';
import type { ValidationIssue } from './validation-issue';

export interface ImportStatus {
  id: string;
  title: string;
  checksum: string;
  status: CurriculumStatus;
  issues: ValidationIssue[];
  publishedVersionId: string | null;
  allowedActions: string[];
  correlationId: string;
}

File: backend/src/RepoFluent.Application/ImportStatus.cs
using RepoFluent.Domain;

namespace RepoFluent.Application;

public sealed record ImportStatus(
    Guid Id,
    string Title,
    string Checksum,
    CurriculumStatus Status,
    IReadOnlyList<ValidationIssue> Issues,
    Guid? PublishedVersionId,
    IReadOnlyList<string> AllowedActions,
    string CorrelationId,
    ValidationReport? ValidationReport,
    WarningAcknowledgement? WarningAcknowledgement,
    string PackageId,
    string PackageVersion,
    int ValidationAttemptCount);
